dds/idl/dds_generator.h

```diff
@@ -1443,6 +1443,50 @@ inline std::string type_kind(AST_Type* type)
   }
 }
 
+inline
+std::string key_only_type_name(const std::string& type_name, FieldFilter field_filter, bool writing)
```

Comment:
This looks like it's duplicating logic in the `RefWrapper`, expect it doesn't look like the `RefWrapper` has `KeyOnly`. I think either it should be using this or the other way around.

Reply:
Are you suggesting to add `KeyOnly` to `RefWrapper`?